From 239df148741e35d5b54749a624f96dfcacc7c57e Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 14 Oct 2024 15:37:57 +0100
Subject: [PATCH 1321/1350] dtoverlays: Fix up imx500 overlays to have unique
 clock nodes

The overlay was creating DT nodes /clocks/clk-aicam and
/clocks/clk-aicam-gated for both cam0 and cam1, which resulted
in one failing.

The clock infrastructure creates the clock name from the node name
without any @N reg extension, so we can't just use that. The nodes
therefore have to be renamed.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/imx500-overlay.dts     | 10 ++++++----
 arch/arm/boot/dts/overlays/imx500-pi5-overlay.dts | 10 ++++++----
 2 files changed, 12 insertions(+), 8 deletions(-)

--- a/arch/arm/boot/dts/overlays/imx500-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx500-overlay.dts
@@ -72,16 +72,16 @@
 		};
 	};
 
-	clocks_frag: fragment@104 {
+	fragment@104 {
 		target-path = "/clocks";
 		__overlay__ {
-			clk_aicam: clk-aicam {
+			clk_aicam: clk-aicam1 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-frequency = <24000000>;
 			};
 
-			clk_aicam_gated: clk-aicam-gated {
+			clk_aicam_gated: clk-aicam-gated1 {
 				compatible = "gpio-gate-clock";
 				clocks = <&clk_aicam>;
 				#clock-cells = <0>;
@@ -98,7 +98,9 @@
 		       <&csi_frag>, "target:0=",<&csi0>,
 			   <&spi_bridge>, "power-supply:0=",<&cam0_reg>,
 		       <&reg_frag>, "target:0=",<&cam0_reg>,
-		       <&cam_node>, "VANA-supply:0=",<&cam0_reg>;
+		       <&cam_node>, "VANA-supply:0=",<&cam0_reg>,
+		       <&clk_aicam>,"name=clk-aicam0",
+		       <&clk_aicam_gated>,"name=clk-aicam-gated0";
 		bypass-cache = <&spi_bridge>,"bypass-cache?";
 	};
 };
--- a/arch/arm/boot/dts/overlays/imx500-pi5-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx500-pi5-overlay.dts
@@ -75,16 +75,16 @@
 		};
 	};
 
-	clocks_frag: fragment@104 {
+	fragment@104 {
 		target-path = "/clocks";
 		__overlay__ {
-			clk_aicam: clk-aicam {
+			clk_aicam: clk-aicam1 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-frequency = <24000000>;
 			};
 
-			clk_aicam_gated: clk-aicam-gated {
+			clk_aicam_gated: clk-aicam-gated1 {
 				compatible = "gpio-gate-clock";
 				clocks = <&clk_aicam>;
 				#clock-cells = <0>;
@@ -103,7 +103,9 @@
 			   <&spi_frag_overlay>, "fast_xfer-gpios:16=35", // CD0_IO1_MICDAT0 (clock)
 			   <&spi_bridge>, "power-supply:0=",<&cam0_reg>,
 		       <&reg_frag>, "target:0=",<&cam0_reg>,
-		       <&cam_node>, "VANA-supply:0=",<&cam0_reg>;
+		       <&cam_node>, "VANA-supply:0=",<&cam0_reg>,
+		       <&clk_aicam>,"name=clk-aicam0",
+		       <&clk_aicam_gated>,"name=clk-aicam-gated0";
 		bypass-cache = <&spi_bridge>,"bypass-cache?";
 	};
 };
